src/encoding.d.ts

```diff
@@ -0,0 +1,20 @@
+import BN from "bn.js";
```

Comment:
Will this conflict with #753 ?

Reply:
You should be able to just remove the `encoding.d.ts` file once `encoding.js` becomes `encoding.ts`.